scripts/bozzcoin.js

```diff
@@ -0,0 +1,203 @@
+      case 'run': return 50;
+      case 'racket steps': return 1 / 18;
+      default: return 0;
+    }
+  }
+
+  function verbToExerciseType(verb) {
+    switch (verb) {
+      case 'ran': return 'run';
+      case 'cycled': return 'cycle';
+      default: return verb;
+    }
+  }
+
+  function bozzExercisesFrequentEnough() {
+    const thresholdInMilliSecs = bozziplierThreshold * 86400000;
+    const bozzLastExercised = robot.brain.get('bozzLastExercised');
+    return (new Date().getTime() - new Date(bozzLastExercised).getTime()) <= thresholdInMilliSecs;
+  }
+
+  // returns true if bozziplier is reset
+  function updateBozziplier(username) {
+    if (username === 'rurouni') {
+      robot.brain.set('bozzLastExercised', new Date());
+      robot.brain.set('bozziplier', 1);
+      return true;
+    }
+    const bozzLastExercised = robot.brain.get('bozzLastExercised');
+    if (!bozzLastExercised || !bozzExercisesFrequentEnough()) {
+      let bozziplier = robot.brain.get('bozziplier');
+      if (!bozziplier) {
+        bozziplier = 1;
+        robot.brain.set('bozzLastExercised', new Date());
+      }
+      const newBozziplier = bozziplier * bozziplierDecay;
+      robot.brain.set('bozziplier', newBozziplier);
+    }
+    return false;
+  }
+
+  function convertToBozzcoin(reps, exerciseType) {
+    return Math.round(reps * earnRate(exerciseType) * robot.brain.get('bozziplier'));
+  }
+
+  function convertToReps(bozzcoin, exerciseType) {
+    return bozzcoin / earnRate(exerciseType);
+  }
+
+  function cheatDayPrices(cheatType) {
+    switch (cheatType) {
+      case 'prata': return 12000;
+      case 'starbucks': return 5000;
+      case 'macs': return 7000;
+      default: return 0;
+    }
+  }
+
+  robot.respond(new RegExp('set bozzcoins to (\\d+)', 'i'), (res) => {
+    if (res.message.user.name === 'lockheed') {
+      const bozzcoinBalance = Number.parseInt(res.match[1], 10);
+      robot.brain.set('bozzcoinBalance', bozzcoinBalance);
+      robot.brain.set('bozzcoinTracker', {});
+      res.send(`Bozzcoin balance reset to ${bozzcoinBalance} :bozzcoin:. Tracker reset.`);
+    }
+  });
+
+  robot.respond(/bozzcoins\?/i, (res) => {
+    const bozzcoinBalance = robot.brain.get('bozzcoinBalance');
+    res.send(`*${bozzcoinBalance}* :bozzcoin:`);
+    res.send(bozzcoinSummaries());
+  });
+
+  robot.respond(new RegExp('did (-?\\d+) (pullups|pushups|situps|squats|lunges|racket steps)', 'i'), (res) => {
+    let repsByUser;
+    const username = res.message.user.name;
+    const repsDone = Number.parseInt(res.match[1], 10);
+    const exerciseType = res.match[2];
+    const bozzcoinTracker = robot.brain.get('bozzcoinTracker');
+    if (username in bozzcoinTracker) {
+      repsByUser = convertToReps(bozzcoinTracker[username], exerciseType);
+    } else {
+      repsByUser = 0;
+    }
+    if (repsDone < (-1 * repsByUser)) {
+      res.send('You can\'t undo more than you have done');
+      return;
+    }
+    switch (exerciseType) {
```

Comment:
maybe this can be abstracted as a validation function

Reply:
Agreed. We can do alot of refactoring. I'm trying to avoid touching the logic part of the code in this PR unless it interfered with the conversion and linting (in case I break anything). We can do further refactoring after we're confident it works?